src/block/block_read.c

```diff
@@ -252,6 +252,16 @@ __wt_block_read_off(WT_SESSION_IMPL *session, WT_BLOCK *block,
 	WT_RET(__wt_read(session, block->fh, offset, size, buf->mem));
 	buf->size = size;
 
+	/*
+	 * Ensure we don't read information that isn't there. It shouldn't ever
+	 * happen, but it's a cheap test.
+	 */
+	if (size < block->allocsize)
```

Comment:
Could you please explain this check for my benefit? I can't make much sense of it.

From reading the surrounding code, my understanding is:
`size`: The size of the data that we want to read from the block. So we want to read beginning from `offset` bytes into the block until `offset + size` bytes into the block.
`block->allocsize`: The size of the block file on disk.

So how come we'd expect that our read (which I'd have thought is a portion of the full block) is larger than the block itself that we're reading from?

Reply:
In this code, I believe `size` is the size in bytes of the block that we're reading from the file.

I think that Coverity is complaining about us reading `size` bytes, then setting `blk` to an offset of about 30 bytes into the bytes we read, and looking at stuff, even if size could have been smaller than the offset. I'm not 100% sure of that, but I think that's the complaint.

I could either test size against `WT_PAGE_HEADER_SIZE` (which is what `WT_BLOCK_HEADER_REF` is using), or against `block.allocsize`, which is minimum size chunk we're allowed to allocate from this file. I couldn't decide which was better, so I just picked one, let me know if that makes things less clear.

Reply:
Yes, Coverity is noticing that you are reading bytes out of a file, and those bytes could technically be anything (say, file corruption or a malicious actor).  So the bytes need to be bounds-checked before using them to calculate addresses from which to read or write memory.